kubespawner/objects.py

```diff
@@ -75,63 +98,86 @@ def make_pod_spec(
         are allowed
     """
-    pod_security_context = {}
-    if run_as_uid is not None:
-        pod_security_context['runAsUser'] = int(run_as_uid)
+    api_client = ApiClient()
+
+    pod = V1Pod()
+    pod.kind = "Pod"
+    pod.api_version = "v1"
+
+    pod.metadata = V1ObjectMeta()
+    pod.metadata.name = name
+    pod.metadata.labels = labels.copy()
+    
+    pod.spec = V1PodSpec()
+
+    security_context = V1PodSecurityContext()
     if fs_gid is not None:
-        pod_security_context['fsGroup'] = int(fs_gid)
-    image_secret = []
+        security_context.fs_group = int(fs_gid)
+    if run_as_uid is not None:
+        security_context.run_as_user = int(run_as_uid)
+    pod.spec.security_context = security_context
+
     if image_pull_secret is not None:
-        image_secret = [{"name": image_pull_secret}]
-    return {
-        'apiVersion': 'v1',
-        'kind': 'Pod',
-        'metadata': {
-            'name': name,
-            'labels': labels,
-        },
-        'spec': {
-            'securityContext': pod_security_context,
-            "imagePullSecrets": image_secret,
-            'containers': [
-                {
-                    'name': 'notebook',
-                    'image': image_spec,
-                    'args': cmd,
-                    'imagePullPolicy': image_pull_policy,
-                    'ports': [{
-                        'containerPort': port,
-                    }],
-                    'resources': {
-                        'requests': {
-                            # If these are None, it's ok. the k8s API
-                            # seems to interpret that as 'no limit',
-                            # which is what we want.
-                            'memory': mem_guarantee,
-                            'cpu': cpu_guarantee,
-                        },
-                        'limits': {
-                            'memory': mem_limit,
-                            'cpu': cpu_limit,
-                        }
-                    },
-                    'env': [
-                        {'name': k, 'value': v}
-                        for k, v in env.items()
-                    ],
-                    'volumeMounts': volume_mounts
-                }
-            ],
-            'volumes': volumes
-        }
-    }
+        pod.spec.image_pull_secrets = []
+        image_secret = V1LocalObjectReference()
+        image_secret.name = image_pull_secret
+        pod.spec.image_pull_secrets.append(image_secret)
+
+    pod.spec.containers = []
+
+    notebook_container = V1Container()
+    notebook_container.name = "notebook"
+    notebook_container.image = image_spec
+    notebook_container.working_dir = working_dir
+    notebook_container.ports = []
+    port_ = V1ContainerPort()
+    port_.name = "notebook-port"
+    port_.container_port = port
+    notebook_container.ports.append(port_)
+    notebook_container.env = []
+    for key, value in env.items():
+        env_ = V1EnvVar()
```

Comment:
This could just be:

```python
notebook_container.env = [V1EnvVar(k, v) for k, v in env.items()]
```

no? https://github.com/kubernetes-incubator/client-python/blob/84bf6cc9dc206cb3314a5b4d2fa1a9c2fa2c5137/kubernetes/client/models/v1_env_var.py#L24 says it takes the properties as params to init